packages/node_modules/node-red/red.js

```diff
@@ -143,7 +143,46 @@ if (process.env.NODE_RED_ENABLE_PROJECTS) {
 }
 
 if (settings.https) {
-    server = https.createServer(settings.https,function(req,res) {app(req,res);});
+    var startupHttps = settings.https;
+    
+    if (typeof startupHttps === "function") {
+        // Get the result of the function, because createServer doesn't accept functions as input
+        startupHttps = startupHttps(); 
+    }  
+
+    server = https.createServer(startupHttps,function(req,res) {app(req,res);});
+    
+    // Refresh https settings at intervals for NodeJs version 11 and above
+    if (settings.httpsRefreshInterval) {
+        if (typeof startupHttps === "function") {        
+            if (server.setSecureContext) {
+                console.log("Refreshing https settings every " + parseInt(settings.credentialRenewalTime) + " seconds.");
```

Comment:
what is this setting ?  credentialRenewalTme is not in settings so will always be null.

Reply:
credentialRenewalTime was the initial name, but Nick used httpsRefreshInterval on Discourse.  So the last change was to rename it.  Don't know where I made this mistake, because I have been testing it after the rename...